extensions/ibm-cos/deployment/src/main/java/org/apache/camel/quarkus/component/ibm/cos/deployment/IbmCosProcessor.java

```diff
@@ -0,0 +1,102 @@
+import com.ibm.cloud.objectstorage.client.builder.AwsClientBuilder;
+import com.ibm.cloud.objectstorage.event.SDKProgressPublisher;
+import com.ibm.cloud.objectstorage.oauth.Token;
+import com.ibm.cloud.objectstorage.regions.AwsRegionProviderChain;
+import com.ibm.cloud.objectstorage.regions.InstanceMetadataRegionProvider;
+import com.ibm.cloud.objectstorage.retry.PredefinedBackoffStrategies;
+import com.ibm.cloud.objectstorage.retry.PredefinedRetryPolicies;
+import com.ibm.cloud.objectstorage.services.s3.internal.AWSS3V4Signer;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageProxyDefinitionBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+
+class IbmCosProcessor {
+
+    private static final String FEATURE = "camel-ibm-cos";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    void runtimeInitialized(BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized) {
+        Stream.of(
+                ClientConfiguration.class.getName(),
+                AwsClientBuilder.class.getName(),
+                AwsRegionProviderChain.class.getName(),
+                InstanceMetadataRegionProvider.class.getName())
+                .map(RuntimeInitializedClassBuildItem::new)
+                .forEach(runtimeInitialized::produce);
+    }
+
+    @BuildStep
+    void reflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
+        // The camel component supports only this AWSS3V4Signer method, might need to register other AbstractAWSSigner subclass if it changes
+        reflectiveClasses.produce(ReflectiveClassBuildItem.builder(AWSS3V4Signer.class).methods().constructors().build());
+
+        reflectiveClasses.produce(ReflectiveClassBuildItem.builder(Token.class).methods().constructors().build());
+
+        // TODO: to remove when https://github.com/IBM/ibm-cos-sdk-java/issues/74 is implemented, released and integrated in Camel
+        reflectiveClasses
+                .produce(ReflectiveClassBuildItem.builder(ScheduledThreadPoolExecutor.class).methods().constructors().build());
+    }
+
+    @BuildStep
+    NativeImageProxyDefinitionBuildItem nativeImageProxyDefinitions() {
+        return new NativeImageProxyDefinitionBuildItem("org.apache.http.conn.HttpClientConnectionManager",
+                "org.apache.http.pool.ConnPoolControl", "com.ibm.cloud.objectstorage.http.conn.Wrapped");
+    }
+
+    @BuildStep
+    void initializeRandomRelatedClassesAtRuntime(
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitializedClasses) {
+        Stream.of(
+                "org.apache.http.impl.auth.NTLMEngineImpl",
```

Comment:
See previous comment about Apache HTTP client.

Reply:
need to try the camel-quarkus-support-httpclient for this one

Reply:
removed by adding the dependency